### ISSUE TITLE
Feature Req: Minimize window prior to hiding

### ISSUE THREAD

**Issue by coderb** [benbuck/rbtray#72]
When using google earth if you click on a "waypoint" it may display a small popup window (child).

It turns out that when using rbtray if you minimize to tray, this popup remains displayed.

However, when you do a normal minimize of google earth the popup is hidden.

Perhaps rbtray should do a normal minimize of the window prior to hiding it and adding the tray icon so that the application gets a chance to handle the minimize windows event and do what it needs.

I've attached some screenshots to illustrate:
![cap1](https://user-images.githubusercontent.com/528515/148301495-8f9a37dc-cf2b-457d-8b1e-38fc84a3fa40.png)
![cap2](https://user-images.githubusercontent.com/528515/148301512-072efe1c-cb48-4c82-ba4a-b2b58e952755.png)

**Comment by dreamforceinc** [comment]
[It's already there](https://github.com/benbuck/rbtray/blob/0bb9dd5dc82507047c900a175afe073acc32bcd7/RBTray.cpp#L109)

**Comment by coderb** [comment]
hi @dreamforceinc - `ShowWindows(SW_HIDE)` makes the window invisible (eg no taskbar entry).  I believe a call to `ShowWindow(SW_MINIMIZE)` prior to the hide would solve my issue (and a corresponding call to `ShowWindow(SW_RESTORE)` when untraying.  thanks.

**Comment by dreamforceinc** [comment]
@coderb 
> I believe a call to `ShowWindow(SW_MINIMIZE)` prior to the hide would solve my issue

Yes, you are right. It works for me, but it have a some unwanted "effects" for some windows. )))
Total Commander now have a two icons in tray after hiding by rbtray. This happens when the option 'Move icon to system tray when minimized' is enabled.
Need to think about it.

**Comment by coderb** [comment]
@dreamforceinc hi- for total commander shouldn't you just unselect the option in total command to minimize to tray?  alternatively, rbtray could check to see if the window is still visible after the minimize and not move to the tray if so (this is probably the most functional solution).

if there is any way i can help to get this issue addressed/merged please let me know.  it is a constant source of annoyance for what is otherwise an outstanding utility.

thanks.

**Comment by benbuck** [comment]
Hi @dreamforceinc and @coderb,

I just responded to pull request #77 by @dreamforceinc.

Thanks,
Benbuck

**Comment by dreamforceinc** [comment]
> if there is any way i can help to get this issue addressed/merged please let me know. it is a constant source of annoyance for what is otherwise an outstanding utility.

Hi @coderb !
Feel free to add your own solution.

**Comment by Chealer** [comment]
@coderb : This is an ITS. As such, tickets are issue reports. Suggestions of solutions/workarounds are welcome in the description, but ticket titles should focus on issues. I suggest a title like:

> Minimizing window may let "sub-window" (popup) displayed (Google Earth, others?)

Also, thanks for indicating at least one affected version and OS version

**Comment by benbuck** [comment]
Hi @coderb, @dreamforceinc, and @Chealer,

Thanks for the suggestion. I've decided not to add any more features to RBTray, so I'm marking this as "Won't Fix". However, I've just today created a pre-release of a program called [Finestray](https://github.com/benbuck/finestray), which can also minimize windows to the tray, and I think might work better for this specific case. Feel free to give it a try to see if it could help for your needs.

Thanks,
Benbuck
